lib/datadog/core/remote/transport/http.rb

```diff
@@ -135,33 +135,6 @@ def default_adapter
             Datadog::Core::Transport::Ext::HTTP::ADAPTER
           end
 
-          def default_hostname(logger: Datadog.logger)
-            logger.warn(
-              'Deprecated for removal: Using #default_hostname for configuration is deprecated and will ' \
-                'be removed on a future ddtrace release.'
-            )
-
-            DO_NOT_USE_ENVIRONMENT_AGENT_SETTINGS.hostname
```

Comment:
Minor: I wonder if `DO_NOT_USE_ENVIRONMENT_AGENT_SETTINGS` still gets used in practice, or if it is a no-op default that always gets replaced by whatever we pass in.

Reply:
When I played around with dependency injection, I did find I believe 1 or 2 places that really still use it, and all the other references were non-functional (value is passed, but never used).